emcc.py

```diff
@@ -1432,7 +1438,11 @@ def phase_setup(options, state, newargs, user_settings):
   if settings.MAIN_MODULE or settings.SIDE_MODULE:
     settings.RELOCATABLE = 1
 
-  if settings.USE_PTHREADS and '-pthread' not in newargs:
+  # Pthreads and Wasm Workers require targeting shared Wasm memory (SAB).
+  if settings.USE_PTHREADS or settings.WASM_WORKERS:
+    settings.SHARED_MEMORY = 1
+
+  if settings.SHARED_MEMORY and '-pthread' not in newargs:
```

Comment:
I don't actually know.

What is needed here is to trigger Clang to enable atomics codegen, TLS, and the multithreading safe memory import code. If `-matomics` `-mbulk-memory` does all of that, then it would be better? It would be desirable not to enable the define `__EMSCRIPTEN_PTHREADS__` when targeting Wasm Workers (which Clang does do with `-pthread` today)

Reply:
Atomics codegen and TLS are enabled by just `-matomics -mbulk-memory`. Whether the linker uses a shared memory in the output module is separately controlled by the `--shared-memory` linker flag, which it looks like is added in builder.py, so that code would have to be updated to check `SHARED_MEMORY` rather than `USE_PTHREADS`. Without `-pthread`, `__EMSCRIPTEN_PTHREADS__` will not be defined.

The only dangerous thing I found when experimenting with this is that with `-pthread`, the stub pthread implementation will be linked in, so e.g. `pthread_mutex_lock` doesn't actually do anything. That's a big enough problem, though, that I'm not sure this is worth pursuing.

Reply:
Updated to use `-matomics -mbulk-memory` instead.

> The only dangerous thing I found when experimenting with this is that with -pthread, the stub pthread implementation will be linked in, so e.g. pthread_mutex_lock doesn't actually do anything. That's a big enough problem, though, that I'm not sure this is worth pursuing.

There will be more work to do to clean away pthreads from Wasm Workers compiled code, but I'll defer that work for later and focus now on getting the "positive" path polished.